fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -994,11 +1024,19 @@ public void testCreateVersionedRDFResource() throws IOException {
             assertEquals("Didn't get a CREATED response!", CREATED.getStatusCode(), getStatus(response));
             checkForVersionedResourceLinkHeader(response);
         }
+        return subjectURI;
     }
 
     private void checkForVersionedResourceLinkHeader(final CloseableHttpResponse response) {
-        assertEquals(1, Arrays.asList(response.getHeaders(LINK)).stream().filter(x -> x.getValue().equals(
-                VERSIONED_RESOURCE_LINK_HEADER)).count());
+        checkForLinkHeader(response,VERSIONED_RESOURCE.getURI(), "type");
+    }
+
+    private void checkForLinkHeader(final CloseableHttpResponse response, final String uri, final String rel) {
+        final int count = (int) Arrays.asList(response.getHeaders(LINK)).stream().filter(x -> {
+            final Link linq = Link.valueOf(x.getValue());
+            return linq.getRel().equals(rel) && linq.getUri().equals(URI.create(uri));
+        }).count();
```

Comment:
😄 I just built a similar function for testing for LDPCv headers, I'm wondering if you build the Link linq outside the filter could you just do `linq.equals(x)`?